services/src/handlers/datasets.rs

```diff
@@ -88,7 +88,7 @@ async fn list_external_datasets_handler<C: Context>(
         .await
         .dataset_provider(&session, provider.into_inner())
         .await?
-        .list(options)
+        .list(options) // TODO: authorization
```

Comment:
If you grab your dataset provider by providing your session, isn't this part of the inner list to use it rather than here?

Reply:
you mena hat the DatasetDB could handle the permissions? I think the DatasetDB should just check whether the user is allowed to access the external provider at all, otherwise fail. And if they are the handler should get the credentials for the given user and provider and put it into the list method.

Reply:
Ah okay, I though calling a dataset provider with a session "primes" it with this session and then you can just call methods on it.